content/en/guide/v10/signals.md

```diff
@@ -0,0 +1,529 @@
+    text.value = "";
+  });
+}
+```
+
+Accessing a signal that has been modified within a batch will reflect its updated value. Accessing a computed signal that has been invalidated by another signal within a batch will re-compute only the necessary dependencies to return an up-to-date value for that computed signal. Any other invalidated signals remain unaffected and are only updated at the end of the batch callback.
+
+```js
+// --repl
+import { signal, computed, effect, batch } from "@preact/signals-core";
+
+const count = signal(0);
+const double = computed(() => count.value * 2);
+const triple = computed(() => count.value * 3);
+
+effect(() => console.log(double.value, triple.value));
+
+batch(() => {
+  // set `count`, invalidating `double` and `triple`:
+  count.value = 1;
+
+  // Despite being batched, `double` reflects the new computed value.
+  // However, `triple` will only update once the callback completes.
+  console.log(double.value); // Logs: 2
+});
+```
+
+
+> :bulb: Tip: Batches can also be nested, in which case batched updates are flushed only after the outermost batch callback has completed.
+
+
+### Rendering optimizations
+
+With signals we can bypass Virtual DOM rendering and bind signal changes directly to DOM mutations. If you pass a signal into JSX in a text position, it will render as text and automatically update in-place without Virtual DOM diffing:
+
+```jsx
+const count = signal(0);
+
+function Unoptimized() {
+  // Re-renders the component when `count` changes:
+  return <p>{count.value}</p>;
+}
+
+function Optimized() {
+  // Text automatically updates without re-rendering the component:
+  return <p>{count}</p>;
+}
+```
+
+To enable this optimization, pass the signal into JSX instead of accessing its `.value` property.
+
+A similar rendering optimization is also supported when passing signals as props on DOM elements.
+
+## API
+
+This section is an overview of the signals API. It's aimed to be a quick reference for folks who already know how to use signals and need a reminder of what's available.
+
+### signal(initialValue)
+
+Creates a new signal with the given argument as its initial value:
+
+```js
+const count = signal(0);
+```
+
+When creating signals within a component, use the hook variant: `useSignal(initialValue)`.
+
+The returned signal has a `.value` property that can be get or set to read and write its value. To read from a signal without subscribing to it, use `signal.peek()`.
+
+### computed(fn)
+
+Creates a new signal that is computed based on the values of other signals. The returned computed signal is read-only, and its value is automatically updated when any signals accessed from within the callback function change.
+
+```js
+const name = signal("Jane");
+const surname = signal("Doe");
+
+const fullName = computed(() => `${name.value} ${surname.value}`);
+```
+
+When creating computed signals within a component, use the hook variant: `useComputed(fn)`.
+
+### effect(fn)
```

Comment:
Should we include a chapter about effect in a component, this feels like a pit fall atm for me atleast